packedForest/src/forestTypes/unsupervisedForests/urerf/urerfTree.h

```diff
@@ -0,0 +1,267 @@
+				}
+
+
+				inline void setAsLeaf(){
+					tree.back().setClass(nodeQueue.back().returnMaxClass());
+					tree.back().setDepth(nodeQueue.back().returnDepth());
+				}
+
+
+				inline void checkOOB(){
+					totalOOB += nodeQueue.back().returnOutSampleSize();
+					correctOOB += nodeQueue.back().returnOutSampleError(tree.back().returnClass());
+				}
+
+
+				inline void makeWholeNodeALeaf(){
+					tree.emplace_back();
+					linkParentToChild();
+					setAsLeaf();
+					checkOOB();
+					leafNodes.emplace_back(nodeQueue.back());
+					//nodeQueue.back().deleteObsIndices();
+					nodeQueue.pop_back();
+				}
+
+				void printTree(){
+					for(auto nd : tree){
+						nd.printNode();
+					}
+				}
+
+				inline void createNodeInTree(){
+					tree.emplace_back();
+					linkParentToChild();
+					tree.back().setCutValue(nodeQueue.back().returnBestCutValue());
+					tree.back().setFeatureValue(nodeQueue.back().returnBestFeature());
+					tree.back().setDepth(nodeQueue.back().returnDepth());
+				}
+
+
+				inline void makeNodeInternal(){
+					createNodeInTree();
+					createChildren();
+				}
+
+				inline bool isLeftNode(){
+					return true;
+				}
+
+				inline bool isRightNode(){
+					return false;
+				}
+
+				inline void createChildren(){
+					nodeQueue.back().moveDataLeftOrRight();
+
+					stratifiedInNodeClassIndicesUnsupervised* leftIndices = nodeQueue.back().returnLeftIndices();
+					stratifiedInNodeClassIndicesUnsupervised* rightIndices = nodeQueue.back().returnRightIndices();
+
+					assert(leftIndices->returnInSampleSize() > 0);
+					assert(rightIndices->returnInSampleSize() > 0);
+
+					int childDepth = nodeQueue.back().returnDepth()+1;
+
+					nodeQueue.pop_back();
+
+					nodeQueue.emplace_back(returnLastNodeID(),childDepth, isLeftNode());
+					nodeQueue.back().loadIndices(leftIndices);
+
+					nodeQueue.emplace_back(returnLastNodeID(),childDepth, isRightNode());
+					nodeQueue.back().loadIndices(rightIndices);
+				}
+
+
+				inline void findTheBestSplit(){
+					nodeQueue.back().findBestSplit();
+				}
+
+
+				inline bool noGoodSplitFound(){
+				        if(nodeQueue.back().returnBestImpurity() == -1)
+                                                return true;
+			
+					return nodeQueue.back().returnBestFeature().empty();
+				}
+
+
+				inline void processANode(){
+					//	timeLogger logTime;
+					//	logTime.startSortTimer();
+					nodeQueue.back().setupNode();
+					//	logTime.stopSortTimer();
+					//	logTime.startGiniTimer();
+					if(shouldProcessNode()){
+						findTheBestSplit();
+						if(noGoodSplitFound()){
+							makeWholeNodeALeaf();
+						}else{
+							makeNodeInternal();
+						}
+					}else{
+						makeWholeNodeALeaf();
+					}
+					//	logTime.stopGiniTimer();
+				}
+
+				inline void processNodes(){
+					while(!nodeQueue.empty()){
+						processANode();
+					}
+				}
+
+				inline void growTree(){
+					loadFirstNode();
+					processNodes();
+				}
+
+		};
+
+}//fp
+#endif //rfTree_h
```

Comment:
urerfTree ?

Reply:
Fixed in 5f65da04650742ed24365cde194085e56b01a60d